app/controllers/admin/works_controller.rb

```diff
@@ -119,7 +121,48 @@ def download_ohms_xml
     send_data @work.oral_history_content!.ohms_xml_text,
       :type => 'text/xml; charset=UTF-8;',
       :disposition => ContentDisposition.format(disposition: "attachment", filename: "#{@work.oral_history_content!.ohms_xml.accession}.xml")
-      #:disposition => "attachment; filename=#{@work.oral_history_content!.ohms_xml.accession}.xml"
+  end
+
+  # PATCH/PUT /admin/works/ab2323ac/submit_ohms_xml
+  def submit_searchable_transcript_source
+    unless params[:searchable_transcript_source].present?
+      redirect_to admin_work_path(@work, anchor: "nav-oral-histories"), flash: { error: "No file received" }
+      return
+    end
+    transcript = params[:searchable_transcript_source].read
+    # Very basic check for now. Not stripping HMTL tags,
+    # even though they really don't belong in these transcripts.
+    #transcript_checks = [true]
+    transcript_checks = [
+      (params[:searchable_transcript_source].content_type.start_with?('text/')),
+      ((MimeMagic.by_magic(transcript).nil?)||(MimeMagic.by_magic(transcript).to_s.start_with? 'text/')),
+      transcript.valid_encoding?
+    ]
+    if transcript_checks.all?
```

Comment:
I don't understand what `all?` with no block does -- is it just the same thing as `present?`?  But maybe we can abolish `transcript_checks` entirely.

Reply:
Just checking that all items in an array are true. No longer relevant though.